crates/core/src/kvs/ds.rs

```diff
@@ -279,127 +279,155 @@ impl Datastore {
 
 	#[allow(unused_variables)]
 	pub async fn new_with_clock(path: &str, clock: Option<Arc<SizedClock>>) -> Result<Datastore> {
+		let (flavour, path) = match path.split_once("://").or_else(|| path.split_once(':')) {
+			None if path == "memory" => ("memory", ""),
+			Some((flavour, path)) => (flavour, path),
+			// Validated already in the CLI, should never happen
+			_ => bail!(Error::Unreachable("Provide a valid database path parameter".to_owned())),
+		};
+
+		let path = if Path::new(path).is_absolute() {
+			// if absolute, remove all slashes except one
+			let normalised = format!("/{}", path.trim_start_matches("/"));
```

Comment:
I think this will cause problems on Windows when mixed with `Path::is_absolute()`. For example `rocksdb:c:\path` or `rocksdb://c:\path` will end up as `rocksdb:/c:\path`. Instead of `Path::new(path).is_absolute()`, should we consider `path.starts_with('/')`? That way we are already sure that the path actually starts with `/`. This would also fix the import issue above.

Reply:
Also, it's possible to not heap allocate a string here but I don't think at this point such optimisations are that important since we are still instantiating the datastore.